api/src/main/java/org/openmrs/api/impl/ConditionServiceImpl.java

```diff
@@ -0,0 +1,127 @@
+import org.openmrs.Patient;
+import org.openmrs.api.APIException;
+import org.openmrs.api.ConditionService;
+import org.openmrs.api.context.Context;
+import org.openmrs.api.db.ConditionDAO;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Date;
+import java.util.List;
+
+/**
+ * This class implements the {@link ConditionService} interface
+ * It defines the methods to handle the condition domain object
+ *
+ * @since 2.2
+ */
+@Transactional
+public class ConditionServiceImpl extends BaseOpenmrsService implements ConditionService{
+
+	private ConditionDAO conditionDAO;
+
+	public void setConditionDAO(ConditionDAO conditionDAO) {
+		this.conditionDAO = conditionDAO;
+	}
+
+	public ConditionDAO getConditionDAO() {
+		return conditionDAO;
+	}
+
+	/**
+	 * Saves a condition
+	 *
+	 * @param condition - the condition to be saved
+	 */
+	@Override
+	public Condition saveCondition(Condition condition) {
+		return conditionDAO.saveCondition(condition);
+	}
+
+	/**
+	 * Gets a condition based on the uuid
+	 *
+	 * @param uuid - uuid of the condition to be returned
+	 * @return the condition
+	 */
+	@Override
+	@Transactional(readOnly = true)
+	public Condition getConditionByUuid(String uuid) {
+		return conditionDAO.getConditionByUuid(uuid);
+	}
+
+	/**
+	 * Gets a condition by id
+	 *
+	 * @param conditionId - the id of the Condition to retrieve
+	 * @return the Condition with the given id, or null if none exists
+	 * @throws APIException
+	 */
+	@Override
+	@Transactional(readOnly = true)
+	public Condition getCondition(Integer conditionId) {
+		return conditionDAO.getCondition(conditionId);
+	}
+
+	/**
+	 * Gets a patient's active conditions
+	 *
+	 * @param patient - the patient to retrieve conditions for
+	 * @return a list of the patient's active conditions
+	 */
+	@Override
+	@Transactional(readOnly = true)
+	public List<Condition> getActiveConditions(Patient patient) {
+		return conditionDAO.getActiveConditions(patient);
+	}
+
+	/**
+	 * Voids a condition
+	 *
+	 * @param condition  - the condition to be voided
+	 * @param voidReason - the reason for voiding the condition
+	 */
+	@Override
+	public Condition voidCondition(Condition condition, String voidReason) {
+		return conditionDAO.saveCondition(condition);
+	}
+
+	/**
+	 * Revive a condition
+	 *
+	 * @param condition Condition to unvoid
+	 * @throws APIException
+	 * @should unset voided bit on given condition
+	 */
+	@Override
+	public Condition unvoidCondition(Condition condition) {
+		return Context.getConditionService().saveCondition(condition);
+	}
+
+	/**
+	 * Completely remove a condition from the database. This should typically not be called
+	 * because we don't want to ever lose data. The data really <i>should</i> be voided and then it
+	 * is not seen in interface any longer (see #voidCondition(Condition) for that one) If other things link to
+	 * this condition, an error will be thrown.
+	 *
+	 * @param condition
+	 * @throws APIException
```

Comment:
Can we remove the above throws?
Can we also remove the should annotation
Remember to also document the condition param